stable_baselines3/dqn/dqn.py

```diff
@@ -0,0 +1,214 @@
+    :param env: (Gym environment or str) The environment to learn from (if registered in Gym, can be str)
+    :param learning_rate: (float or callable) The learning rate, it can be a function
+        of the current progress (from 1 to 0)
+    :param buffer_size: (int) size of the replay buffer
+    :param learning_starts: (int) how many steps of the model to collect transitions for before learning starts
+    :param batch_size: (int) Minibatch size for each gradient update
+    :param gamma: (float) Discount factor
+    :param train_freq: (int) Update the model every ``train_freq`` steps.
+    :param gradient_steps: (int) How many gradient update after each step
+    :param tau: (float) the soft update coefficient ("polyak update", between 0 and 1)
+    :param target_update_interval: (int) update the target network every ``target_update_interval`` steps.
+    :param exploration_fraction: (float) fraction of entire training period over which the exploration rate is reduced
+    :param exploration_initial_eps: (float) initial value of random action probability
+    :param exploration_final_eps: (float) final value of random action probability
+    :param max_grad_norm: (float) The maximum value for the gradient clipping
+    :param tensorboard_log: (str) the log location for tensorboard (if None, no logging)
+    :param create_eval_env: (bool) Whether to create a second environment that will be
+        used for evaluating the agent periodically. (Only available when passing string for the environment)
+    :param policy_kwargs: (dict) additional arguments to be passed to the policy on creation
+    :param verbose: (int) the verbosity level: 0 no output, 1 info, 2 debug
+    :param seed: (int) Seed for the pseudo random generators
+    :param device: (str or th.device) Device (cpu, cuda, ...) on which the code should be run.
+        Setting it to auto, the code will be run on the GPU if possible.
+    :param _init_setup_model: (bool) Whether or not to build the network at the creation of the instance
+    """
+
+    def __init__(self, policy: Union[str, Type[DQNPolicy]],
+                 env: Union[GymEnv, str],
+                 learning_rate: Union[float, Callable] = 2.5e-4,
+                 buffer_size: int = 1000000,
+                 learning_starts: int = 50000,
+                 batch_size: Optional[int] = 32,
+                 gamma: float = 0.99,
+                 train_freq: int = 4,
+                 gradient_steps: int = 1,
+                 tau: float = 1.0,
+                 target_update_interval: int = 10000,
+                 exploration_fraction: float = 0.02,
+                 exploration_initial_eps: float = 1.0,
+                 exploration_final_eps: float = 0.1,
+                 max_grad_norm: float = 10,
+                 tensorboard_log: Optional[str] = None,
+                 create_eval_env: bool = False,
+                 policy_kwargs: Optional[Dict[str, Any]] = {'optimizer_class': th.optim.RMSprop,
```

Comment:
you should not use mutable as default arguments... please look at how it is done for A2C